src/background/services/pier-service.ts

```diff
@@ -63,6 +66,13 @@ export class PierService {
         this.urbitPath = joinPath(binariesPath, 'urbit');
         this.resumesInProgress = new Map();
         this.recoverBootingShips();
+        
+        this.getPiers().then(piers => {
+            ipcRenderer.invoke('piers', piers);
+        })
+        setInterval(async () => {
+            ipcRenderer.invoke('piers', await this.getPiers());
+        }, 3000);
```

Comment:
Feels a bit like a hack. Was it tricky to roll this communication into the `addPier`/`updatePier` methods for some reason? Or maybe there are other touch points for pier modifications I'm not considering?

Reply:
It is, I thought about add/update and deferred, but this thinking was in the back of my head. I'll try adding it to see if it's problematic, but would feel better for sure